# Introduction

### Introduction or week 1 work

After opening the file the 1st thing i noticed it that we need to manage the upper portion of the bok having title and oher things etc.

In [2]:
f = open("PeterPan.txt", "r", encoding="utf-8-sig")
t = f.read()
len(t)

274951

The charater length of the text in the book is 274951 which means there are total 274951 characters in the whole book.

# NLTK TOKANIZERS

## Split the book to sentences

In [3]:
import nltk.tokenize as tok
sents = tok.sent_tokenize(t, language= "english")
len(sents)

2563

Yes the number corrosponds to the number of my expectations.

In [4]:
print(sents[0])
print ("------------------")
print(sents[1])
print ("------------------")


The Project Gutenberg eBook of Peter Pan, by James M. Barrie

This eBook is for the use of anyone anywhere in the United States and
most other parts of the world at no cost and with almost no restrictions
whatsoever.
------------------
You may copy it, give it away or re-use it under the terms
of the Project Gutenberg License included with this eBook or online at
www.gutenberg.org.
------------------


In [5]:
import re
t = re.sub(r"\n+", " ", t) #re.sub() is subsitude method in regular expression.
print(t[:100])

The Project Gutenberg eBook of Peter Pan, by James M. Barrie This eBook is for the use of anyone any


In [6]:
tokens = tok.word_tokenize(t, language="english")
len(tokens)

61946

In [7]:
f = open("peterpan.txt", "r", encoding="utf-8-sig")
t = f.read()
tokens = tok.word_tokenize(t, language="english")
print(len(t))
print(tokens[:30])


274951
['The', 'Project', 'Gutenberg', 'eBook', 'of', 'Peter', 'Pan', ',', 'by', 'James', 'M.', 'Barrie', 'This', 'eBook', 'is', 'for', 'the', 'use', 'of', 'anyone', 'anywhere', 'in', 'the', 'United', 'States', 'and', 'most', 'other', 'parts', 'of']


# Frequency distribution and pandas

In [8]:
import collections
c = collections.Counter(tokens) # The Counter class is used to create a dictionary-like object 
del c["and"]
print(c.most_common(10))
print(c["with"])

[(',', 3981), ('.', 2441), ('the', 2330), ('“', 1464), ('”', 1464), ('to', 1222), ('a', 959), ('of', 923), ('was', 920), ('he', 878)]
361


In [9]:
import string
string.punctuation

'!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~'

In [10]:
def remove_punctuation(text):
    translator = str.maketrans("", "", string.punctuation)
    return text.translate(translator)
clean_text= remove_punctuation(t)

In [11]:
len(clean_text)
tokens = tok.word_tokenize(clean_text, language="english")
c = collections.Counter(tokens) # The Counter class is used to create a dictionary-like object 
print(c.most_common(10))
print(c["with"])

[('the', 2329), ('“', 1464), ('”', 1464), ('and', 1420), ('to', 1222), ('a', 959), ('of', 923), ('was', 923), ('he', 878), ('in', 698)]
361


In [21]:
import pandas as pd
data = c.most_common()
print (data)
df= pd.DataFrame(data, columns= ["Word","Frequency"])
print(df)

[('the', 2329), ('“', 1464), ('”', 1464), ('and', 1420), ('to', 1222), ('a', 959), ('of', 923), ('was', 923), ('he', 878), ('in', 698), ('it', 661), ('that', 609), ('’', 558), ('had', 499), ('you', 480), ('she', 477), ('they', 476), ('I', 464), ('her', 456), ('his', 456), ('not', 414), ('Peter', 402), ('but', 391), ('for', 382), ('with', 361), ('is', 359), ('said', 358), ('Wendy', 357), ('on', 351), ('him', 338), ('at', 318), ('as', 315), ('s', 280), ('them', 279), ('have', 258), ('be', 255), ('all', 248), ('were', 246), ('this', 227), ('so', 215), ('their', 213), ('would', 210), ('one', 201), ('are', 200), ('by', 188), ('He', 181), ('It', 174), ('The', 166), ('t', 163), ('out', 163), ('up', 160), ('no', 154), ('when', 154), ('Hook', 153), ('we', 148), ('from', 146), ('there', 143), ('could', 142), ('or', 137), ('cried', 136), ('been', 136), ('who', 135), ('John', 133), ('if', 132), ('now', 128), ('which', 126), ('She', 126), ('did', 123), ('do', 123), ('time', 120), ('Darling', 115), 

In [22]:
tokens = tok.word_tokenize(clean_text, language="english")
print("Tokens= ", len(tokens))
print("Type of tokens= ", len(set(tokens))) # convert list of words to set leavinh just unique words.
print("Ratio= ",len(tokens)/len(set(tokens)))

Tokens=  54354
Type of tokens=  5836
Ratio=  9.313570938999314


In [23]:
df[df["Frequency"]>=1000]

Word  Frequency
0  the       2329
1    “       1464
2    ”       1464
3  and       1420
4   to       1222

In [28]:
h= df[df["Frequency"]==1]
hapex= len(h)
print(h)

            Word  Frequency
2903       Title          1
2904      Author          1
2905     Release          1
2906        Date          1
2907        July          1
...          ...        ...
5831    facility          1
5832    includes          1
5833     produce          1
5834   subscribe          1
5835  newsletter          1

[2933 rows x 2 columns]


In [32]:
percentage= hapex/len(tokens)*100
print(percentage)

5.396107002244545


5.3% of the words are hapex words in the whole document.

In [35]:
count=0
for words in tokens:
    if words[0]=="A" and words.istitle():
        count= count+1
print(count)

261
